lib/compilers/ppci.js

```diff
@@ -0,0 +1,20 @@
+
+const Compile = require('../base-compiler'),
+    exec = require('../exec');
+
+function compilePpci(info, env) {
+    const compiler = new Compile(info, env);
+
+    compiler.exec = function (compiler, args, options) {
```

Comment:
I just thought that `compiler.runCompiler` is a better function to use in this case, rather than the exec one. For some examples, [here is how it's defined in the base-compiler](https://github.com/mattgodbolt/compiler-explorer/blob/master/lib/base-compiler.js#L96) and [a pascal override](https://github.com/mattgodbolt/compiler-explorer/blob/master/lib/compilers/pascal.js#L162) so you get an idea

Reply:
This would be ideal, but then the version check will fail, as that only uses exec(), unless you also override initialise() of course, but that will be a lot more complicated and messy I think.

(Maybe the Initialise() should be refactored in general, but that'll be easier when the es6 merge happens)

Reply:
Indeed the version check failed. And this way, the code is as small as possible and still works. runCompiler is a somewhat bigger function.

Reply:
I noticed another problem with using exec for this, exec is also used for objdump and demangle, which are not python programs obviously.

So I think we could do 2 things, override runCompiler and initialise like suggested, or I was thinking we could do a similar trick we do for Wine applications, is put the full path to the .py file in properties and if .py, then unwrap that and use the custom exec implementation.

Reply:
* actually just wrote that and found out ppci doesn't actually output anything binary?

Reply:
The output is assembly with the -S option. Otherwise you will get a custom oj format which is not ELF so objdump will fail on that in a horrible way. Should I modify exec such that only when python code is executed python is invoked.

Reply:
Just modified the exec method to check if python is required. The code works, but it still feels like a hack.

Reply:
@windelbouwman I sent a PR to indicate what I was thinking about.

Reply:
Even better! I think this is the best option so far. This is then even flexible for ppci.cli.pascal.py as well!

Reply:
If there are plans to add a pascal ppci, I think C's ppci compilerType should be `ppcic` to be able to distinguish from the future `ppcip` for pascal

Reply:
That is, if they end up needing different types!

Reply:
I just tried it out, but unfortunately, the Pascal compiler is extremely minimal at the moment. Definitely can't compile the examples yet. :(

Reply:
Pascal support is in very early stage. To be continued in future times.